weave/ops_arrow/list_ops.py

```diff
@@ -888,10 +888,46 @@ def flatten_return_type(input_types):
 def flatten(arr):
     # TODO:
     #   - handle N levels instead of 1
-    #   - handle tags
+
     arrow_data = arr._arrow_data
     if is_list_arrowweavelist(arr):
-        arrow_data = arrow_data.flatten()
+        # unwrap tags
+
+        tags = None
+        if isinstance(arr.object_type, tagged_value_type.TaggedValueType):
+            value_awl, tags_awl = (
+                arr.tagged_value_value(),
+                arr.tagged_value_tag(),
+            )
+
+            values = value_awl._arrow_data
+            tags = tags_awl._arrow_data
+
+            tag_arrays = []
+
+            # while this technically calls to_pylist(), it is not actually
+            # breaking any data out to python in our internal weave definition,
+            # because it is just a list of integers that are the list offsets
+            # that are passed to the pa.repeat() call below
+
+            num_repeats = pc.subtract(
+                values.offsets[1:], values.offsets[:-1]
+            ).to_pylist()
+
+            for tag, rnum in zip(tags, num_repeats):
```

Comment:
Doesn't this end up doing what flatten does? I'm confused, might need a hint.

Reply:
ok, well we still can't use to_pylist :). need to figure out how to do this in arrow. I can help here

Reply:
Updated to keep things in arrow following https://github.com/apache/arrow/issues/27923#issuecomment-1378008351